apps/quick-learn-backend/src/routes/lesson-progress/lesson-progress.controller.ts

```diff
@@ -46,22 +46,11 @@ export class LessonProgressController {
     return new SuccessResponse('Course completed lessons', data);
   }
 
-  // @Get(':courseId/progresscount')
-  // async getLessonProgressCount(
-  //   @Param('courseId') courseId: number,
-  //   @Request() req,
-  // ) {
-  //   return await this.lessonProgressService.getCourseLessonCount(
-  //     req.user.id,
-  //     courseId,
-  //   );
-  // }
-
-  @Get('/userprogress')
-  async getAllUserProgress(@Request() req) {
+  @Get('/userprogress:userID?')
+  async getAllUserProgress(@Request() req, @Param('userID') userID?: number) {
```

Comment:
User current user decorator to get the user rather than extracting from the request body.